src/frequenz/client/microgrid/component/_component_proto.py

```diff
@@ -0,0 +1,562 @@
+            message,
+        )
+
+    return component
+
+
+class ComponentBaseData(NamedTuple):
+    """Base data for a component, extracted from a protobuf message."""
+
+    component_id: ComponentId
+    microgrid_id: MicrogridId
+    name: str | None
+    manufacturer: str | None
+    model_name: str | None
+    category: ComponentCategory | int
+    status: ComponentStatus | int
+    lifetime: Lifetime
+    rated_bounds: dict[Metric | int, Bounds]
+    category_specific_metadata: dict[str, Any]
+    category_mismatched: bool = False
+
+
+def component_base_from_proto_with_issues(
+    message: components_pb2.Component,
+    *,
+    major_issues: list[str],
+    minor_issues: list[str],
+) -> ComponentBaseData:
+    """Extract base data from a protobuf message and collect issues.
+
+    Args:
+        message: The protobuf message.
+        major_issues: A list to append major issues to.
+        minor_issues: A list to append minor issues to.
+
+    Returns:
+        A `ComponentBaseData` named tuple containing the extracted data.
+    """
+    component_id = ComponentId(message.id)
+    microgrid_id = MicrogridId(message.microgrid_id)
+
+    name = message.name or None
+    if name is None:
+        minor_issues.append("name is empty")
+
+    manufacturer = message.manufacturer or None
+    if manufacturer is None:
+        minor_issues.append("manufacturer is empty")
+
+    model_name = message.model_name or None
+    if model_name is None:
+        minor_issues.append("model_name is empty")
+
+    status = enum_from_proto(message.status, ComponentStatus)
+    if status is ComponentStatus.UNSPECIFIED:
+        major_issues.append("status is unspecified")
+    elif isinstance(status, int):
+        major_issues.append("status is unrecognized")
+
+    lifetime = _get_operational_lifetime_from_proto(
+        message, major_issues=major_issues, minor_issues=minor_issues
+    )
+
+    rated_bounds = _metric_config_bounds_from_proto(
+        message.metric_config_bounds,
+        major_issues=major_issues,
+        minor_issues=minor_issues,
+    )
+
+    category = enum_from_proto(message.category, ComponentCategory)
+    if category is ComponentCategory.UNSPECIFIED:
+        major_issues.append("category is unspecified")
+    elif isinstance(category, int):
+        major_issues.append(f"category {category} is unrecognized")
+
+    metadata_category = message.category_type.WhichOneof("metadata")
+    category_specific_metadata: dict[str, Any] = {}
+    if metadata_category is not None:
+        category_specific_metadata = MessageToDict(
+            getattr(message.category_type, metadata_category),
+            always_print_fields_with_no_presence=True,
+        )
+
+    category_mismatched = False
+    if (
+        metadata_category
+        and isinstance(category, ComponentCategory)
+        and category.name.lower() != metadata_category
+    ):
+        major_issues.append("category_type.metadata does not match the category_type")
```

Comment:
maybe also add the metadata category found here, because that info is not easily available from elsewhere?

Reply:
When issues are found, the original protobuf message will be printed in full, this is why I make issue strings very short.